addons/payment/models/payment_acquirer.py

```diff
@@ -129,6 +130,21 @@ def _valid_field_parameter(self, field, name):
     show_done_msg = fields.Boolean(compute='_compute_view_configuration_fields')
     show_cancel_msg = fields.Boolean(compute='_compute_view_configuration_fields')
 
+    # Restrain the partner by domain.
+    payment_domain = fields.Char(string="Restraint Partners on")
+
+    # To restraint acquirer in function of a maximal amount to be paid
+    # (e.g. Paypal only handles amounts less than 1000).
+    maximal_amount = fields.Float(help="Leave as '0.00' to ignore a maximal amount.")
+
+    # Needed to get the the payment domain as a valid domain
+    def is_allowed_partner(self, partner_id):
+        return (not self.payment_domain or self.payment_domain == "[]") or (
+                partner_id and partner_id in
+                [p['id'] for p in
+                 self.env['res.partner'].search(ast.literal_eval(self.payment_domain))]
```

Comment:
it is ugly, because the result of the partner search can return million of partner.
I have add sudo to allow read restricted field.

```python
def is_allowed_partner(self, partner_id):
      self.ensure_one()
      return bool(self.env['res.partner'].sudo().browse(partner_id).filtered_domain(safe_eval(self.payment_domain or '[]')))
```

Reply:
thank you for the suggestion, I will implement some of it